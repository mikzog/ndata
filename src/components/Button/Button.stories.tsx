import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button from './Button';

export default {
  title: 'Button',
};

export const text = () => (
  <Button onClick={action('clicked')}>Hello Button</Button>
);

export const emoji = () => (
  <Button onClick={action('clicked')}>
    <span role="img" aria-label="so cool">
      😀 😎 👍 💯
    </span>
  </Button>
);

storiesOf('Button', module)
  .add('with text', text)
  .add('with emoji', emoji);
