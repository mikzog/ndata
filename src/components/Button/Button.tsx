import React from 'react';

interface Props {
  onClick?: () => void;
}

export const Button: React.FC<Props> = ({ children, onClick }) => {
  return <button onClick={onClick}>{children}</button>;
};

export default Button;
