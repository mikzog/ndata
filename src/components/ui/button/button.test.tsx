import React from 'react';
import renderer from 'react-test-renderer';
import Button from './button';

describe('Button', () => {
  test('renders without crashing', () => {
    const component = renderer.create(<Button />);
    const json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  test('renders without loading', () => {
    const component = renderer.create(<Button loading />);
    const json = component.toJSON();

    expect(json).toMatchSnapshot();
  });
});
