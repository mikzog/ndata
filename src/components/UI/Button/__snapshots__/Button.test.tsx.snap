// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<button
  className="nd-button"
/>
`;

exports[`Button renders without loading 1`] = `
<div
  className="nd-button-loading load-inline"
>
  <button
    className="nd-button"
  >
    <div
      className="nd-loader show small"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </button>
</div>
`;
